src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -664,6 +618,33 @@ public override string ToString()
             return Value ?? string.Empty;
         }
 
+        private static CompareInfo GetCompareInfo(StringComparison comparisonType, out CompareOptions compareOptions)
+        {
+            switch (comparisonType)
+            {
+                case StringComparison.CurrentCulture:
+                case StringComparison.CurrentCultureIgnoreCase:
+                    compareOptions = (CompareOptions)((int)comparisonType & 1);
```

Comment:
(nit) This is kind of tricky, and not really intuitive to a reader. Is it written this way for performance?

A more straight forward way of writing it would be:

```C#

            switch (comparisonType)
            {
                case StringComparison.CurrentCulture:
                    compareOptions = CompareOptions.None;
                    return CultureInfo.CurrentCulture.CompareInfo;

                case StringComparison.CurrentCultureIgnoreCase:
                    compareOptions = CompareOptions.IgnoreCase;
                    return CultureInfo.CurrentCulture.CompareInfo;
```

Reply:
Yeah, it was roughly copied from the switch statements [in string](https://github.com/dotnet/runtime/blob/8a52f1e948b6f22f418817ec1068f07b8dae2aa5/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs#L654). But after thinking through this further, I think those methods were trying to minimize codegen size, while we don't have that exact consideration here. We should be able to get away with the readable thing rather than the smallest-codegen thing.